backend/src/main/java/nl/itvitae/twitbook/security/AuthController.java

```diff
@@ -30,29 +32,47 @@ public class AuthController {
   private final AuthenticationManager authenticationManager;
   private final JWTService jwtService;
   private final PasswordEncoder passwordEncoder;
+  private static final int MIN_USERNAME_LENGTH = 3;
+  private static final int MAX_USERNAME_LENGTH = 25;
 
   @PostMapping("login")
   public ResponseEntity<?> login(@RequestBody LoginModel loginModel) {
     try {
       Authentication authentication = authenticationManager.authenticate(
-        new UsernamePasswordAuthenticationToken(loginModel.username(), loginModel.password()));
+          new UsernamePasswordAuthenticationToken(loginModel.username(), loginModel.password()));
 
-      if (!authentication.isAuthenticated())
+      if (!authentication.isAuthenticated()) {
         return new ResponseEntity<>("Invalid user credentials", HttpStatus.UNAUTHORIZED);
+      }
 
       return new ResponseEntity<>(jwtService.generateUserJWT(loginModel.username()), HttpStatus.OK);
-    }
-    catch(AuthenticationException exc) {
+    } catch (AuthenticationException exc) {
       return new ResponseEntity<>("Invalid user credentials", HttpStatus.UNAUTHORIZED);
     }
   }
 
   @PostMapping("register")
-  public ResponseEntity<?> createUser(@RequestBody RegisterModel model, UriComponentsBuilder uriBuilder) {
-    if (userRepository.findByUsernameIgnoreCase(model.username()).isPresent())
+  public ResponseEntity<?> createUser(@RequestBody RegisterModel model,
+      UriComponentsBuilder uriBuilder) {
+
+    final Pattern pattern = Pattern.compile(
+        String.format("[\\w\\d_]{%d,%d}", MIN_USERNAME_LENGTH, MAX_USERNAME_LENGTH),
```

Comment:
Good! Please add this Regex pattern to Globals.ts in the frontend and use it as `<input pattern={Globals.usernamePattern} />` in the RegisterComponent